File: katas/Palindrom/solutions/fricke_frederik/CSharp/Palindrome/PalindromeTest/PalindromeTest.cs
using NUnit.Framework;

namespace PalindromeTest;
using Palindrome;

public class Tests
{
    private Palindrome _palindrome;
    
    [SetUp]
    public void Setup()
    {
        _palindrome = new Palindrome();
    }

    [Test]
    public void SimplePalindromeTest()
    {
        Assert.True(_palindrome.IsPalindrome("Abba"));
        Assert.True(_palindrome.IsPalindrome("Lagerregal"));
        Assert.True(_palindrome.IsPalindrome("Reliefpfeiler"));
        Assert.True(_palindrome.IsPalindrome("Rentner"));
        Assert.True(_palindrome.IsPalindrome("Dienstmannamtsneid"));
        Assert.False(_palindrome.IsPalindrome("Test"));
    }

    [Test]
    public void SentencePalindromeTest()
    {
        Assert.True(_palindrome.IsPalindrome("Tarne nie deinen Rat!"));
        Assert.True(_palindrome.IsPalindrome("Eine güldne, gute Tugend: Lüge nie!"));
        Assert.True(_palindrome.IsPalindrome("Ein agiler Hit reizt sie. Geist?! Biertrunk nur treibt sie. Geist ziert ihre Liga nie!"));
        Assert.False(_palindrome.IsPalindrome("Dies ist ein einfacher Test für eine Falsch-Prüfung"));
    }
}

File: katas/Palindrom/solutions/fricke_frederik/Java/src/de/fricke/test/PalindromeTest.java
package de.fricke.test;

import de.fricke.Palindrome;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class PalindromeTest {
    @Test
    public void SimplePalindromeTest()
    {
        Palindrome palindrome = new Palindrome();
        assertTrue(palindrome.IsPalindrome("abba"));
        assertTrue(palindrome.IsPalindrome("Lagerregal"));
        assertTrue(palindrome.IsPalindrome("Reliefpfeiler"));
        assertTrue(palindrome.IsPalindrome("Rentner"));
        assertTrue(palindrome.IsPalindrome("Dienstmannamtsneid"));
        assertFalse(palindrome.IsPalindrome("abcde"));
    }

    @Test
    public void ExtendedPalindromeTest()
    {
        Palindrome palindrome = new Palindrome();
        assertTrue(palindrome.IsPalindrome("Tarne nie deinen Rat!"));
        assertTrue(palindrome.IsPalindrome("Eine güldne, gute Tugend: Lüge nie!"));
        assertTrue(palindrome.IsPalindrome("Ein agiler Hit reizt sie. Geist?! Biertrunk nur treibt sie. Geist ziert ihre Liga nie!"));
        assertFalse(palindrome.IsPalindrome("Eine einfache Falch-Prüfung"));
    }
}
